### PR TITLE
Add docs showing pip-tools usage on dependencies

### PR DESCRIPTION
**Pull request by ericholscher**
This will help users not have random jinja or docutils updates break their builds.

Fixes https://github.com/readthedocs/readthedocs.org/issues/9039

### CODE REVIEW

File: docs/user/guides/reproducible-builds.rst [hunk 1]
```diff
@@ -182,3 +182,89 @@ or our Conda docs about :ref:`environment files <guides/conda:creating the \`\`e
...
+
+✅ Good:
+    All your transitive dependencies will stay defined,
+    not allowing a package upgrade to break your docs.
+
+   .. code-block::
+
+      # File: docs/requirements.in
+
+      sphinx==4.2.0
```
Comment (agjohnson) [inline docs/user/guides/reproducible-builds.rst:208]: Captions show up a bit nicer, and outside the code example.

```suggestion
   .. code-block::
      :caption: docs/requirements.in

      sphinx==4.2.0
```

File: docs/user/guides/reproducible-builds.rst [hunk 2]
```diff
@@ -182,3 +182,89 @@ or our Conda docs about :ref:`environment files <guides/conda:creating the \`\`e
...
+
+   .. code-block:: yaml
+
+      # File: docs/requirements.txt
+
+      # This file is autogenerated by pip-compile with python 3.7
+      # To update, run:
```
Comment (agjohnson) [inline docs/user/guides/reproducible-builds.rst:215]:
```suggestion
   .. code-block:: yaml
      :caption: docs/requirements.txt

      # This file is autogenerated by pip-compile with python 3.7
      # To update, run:
```

File: docs/user/guides/reproducible-builds.rst [hunk 3]
```diff
@@ -182,3 +182,89 @@ or our Conda docs about :ref:`environment files <guides/conda:creating the \`\`e
...
+
+We recommend `pip-tools`_ to help address this problem.
+It allows you to specify a ``requirements.in`` file with your first-level dependencies,
+and it generates a ``requirementes.txt`` file with the full set of transitive dependencies.
+
+.. _pip-tools: https://pip-tools.readthedocs.io/en/latest/
+
+✅ Good:
+    All your transitive dependencies will stay defined,
+    not allowing a package upgrade to break your docs.
```
Comment (agjohnson) [inline docs/user/guides/reproducible-builds.rst:202]: A few more clarity updates

```suggestion
    All your transitive dependencies will stay defined,
    which ensures new package releases will not break your docs.
```

File: docs/user/guides/reproducible-builds.rst [hunk 4]
```diff
@@ -160,7 +160,7 @@ Some examples:
       readthedocs-sphinx-search
 
    .. code-block:: yaml
-      
+
       # File: docs/environment.yaml
```
Comment (agjohnson) [inline docs/user/guides/reproducible-builds.rst:164]: Also `:caption:` here as well.

**Review by agjohnson — COMMENTED** [review]
It's probably also worth noting to users that there is specific tooling that does this job much better, like Poetry and friends with lock files. Our suggestion is just the easiest to migrate to.

**Review by humitos — COMMENTED** [review]
This is a good recommendation, I'm 👍🏼 on merging it.

However, I think the "Pinning transitive dependencies" section that you are adding is very similar to the previous one "Pinning dependencies" that does not pin transitive dependencies. So, I'd say that we should either:

* refer to the previous section communicating that pinning all the versions, including transitive dependencies is better than the previous section but it implies an extra step
* merge the two sections into one and only recommend pinning with pip-tools

Besides, this new section does not have a " Bad" subsection as the other have.

**Comment by ericholscher** [comment]
Yea, there's definitely some additional things to work on here, this is just a basic start.

**Pull request merged**